### PR TITLE
Update .gitignore

### PR DESCRIPTION
**Pull request by OoLunar**
# Summary
Updates the `.gitignore` file and adds the `.vscode` folder.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -397,11 +400,81 @@ FodyWeavers.xsd
...
+*.msm
+*.msp
 
-# Github Codespaces configuration file
-.devcontainer/devcontainer.json
+# Windows shortcuts
+*.lnk
 
-# Docfx
-docs/_site
```
Comment (akiraveliara) [inline .gitignore:407]: are we sure we want to un-ignore docfx?

File: .gitignore [hunk 2]
```diff
@@ -397,11 +400,81 @@ FodyWeavers.xsd
 # JetBrains Rider
 *.sln.iml
 
-# Credentials for the DSharpPlus.Test bot
-DSharpPlus.Test/config.json
```
Comment (akiraveliara) [inline .gitignore:401]: this is dangerous. we absolutely should not do this.

**Review by akiraveliara — CHANGES_REQUESTED** [review]
few potentially problematic changes

File: .gitignore (hunk 1 above)
Comment (InFTord) [reply .gitignore:407]: do NOT remove docfx from gitignore, we don't want JS in our repo

File: .gitignore (hunk 2 above)
Comment (OoLunar) [reply .gitignore:401]: Honestly I think we should just create a better test bot. But yeah I'll readd that

File: .gitignore (hunk 1 above)
Comment (OoLunar) [reply .gitignore:407]: M'bad, thought that was included by default

**Review by akiraveliara — APPROVED** [review]

**Pull request merged**